user/config-imports.md

```diff
@@ -0,0 +1,203 @@
+across many repositories, you can define them in a shared snippet:
+
+```yaml
+rvm:
+  - 2.5
+  - 2.6
+```
+{: data-file="rubies.yml"}
+
+You can then `import` that snippet into your `.travis.yml`. The following
+configuration imports the file `rubies.yml` from the `main` branch of the
+`shared-configs` repository of the `travis-ci` account:
+
+```yaml
+import: travis-ci/shared-configs:rubies.yml@main
+script: bundle exec rake
+```
+{: data-file=".travis.yml"}
+
+Resulting in the following configuration:
+
+```yaml
+rvm:
+  - 2.5
+  - 2.6
+script: bundle exec rake
+```
+{: data-file=".travis.yml"}
+
+## Import sources
+
+The format of the `import` source is `<account>/<repository>:<path>@<ref>` in
+which `<ref>` can be any valid Git reference, such as a commit sha, branch name,
+or tag name.
+
+> Public repositories can import sources from public repositories, but not
+> private repositories. Private repositories can import sources from both
+> public and private repositories. See [private repositories](#private-repositories)
+> for more information.
+
+Import a single source:
+
+```yaml
+import: travis-ci/build-configs:rubies.yml@main
+
+# or
+
+import:
+  source: travis-ci/build-configs:rubies.yml@main
+```
+{: data-file=".travis.yml"}
+
+Import multiple sources:
+
+```yaml
+import:
+  - travis-ci/build-configs:rubies.yml@adf1235
+  - travis-ci/build-configs:other.yml@v1
+
+# or
+
+import:
+  - source: travis-ci/build-configs:rubies.yml@adf1235
+  - source: travis-ci/build-configs:other.yml@v1
+```
+{: data-file=".travis.yml"}
+
+By default, if the same keys are present in multiple imported snippets, they
+are overwritten with the last configuration imported. But you can also specify
+[deep_merge](#merge-mode).
+
+## Importing configs from the same repository
+
+When importing configurations stored in the same repository as your
+`travis.yml`, you can ommit the `<account>/<repository>` part:
+
+```yaml
+# local imports fetch the same git commit ref
+import:
+- one.yml
+- path/to/other.yml
+```
+{: data-file=".travis.yml"}
+
+The path is relative to the repository's root.
+
+## Importing specific versions of configs
+
+
+Configurations imported from a different repository, fetch the fetch the latest
+version of the default branch in the repository.
+
+Configurations imported from the same repository, fetch the commit you are
+currently building by default. This is intended to help while you are creating
+and testing the shared configurations.
+
+You can specify the exact version of a config snippet by using any valid Git
+reference:
+
+```yaml
+import:
+- one.yml@production
+- travis-ci/build-configs/other.yml@v1.0.0
+```
+{: data-file=".travis.yml"}
+
+## Merge modes
+
+The merge mode controls how imported configs are being merged (combined) into
+the importing config. A different merge mode can be specified for each imported
+config source.
+
+There are these merge modes:
```

Comment:
```suggestion
These merge modes are possible:
```

Reply:
Hmmmm ... `possible` reads odd to me here. they're not just "possible", but these are all the defined values. i'll think about this more.